server/src/main/java/org/elasticsearch/env/BuildVersion.java

```diff
@@ -58,6 +62,11 @@ public abstract class BuildVersion {
      */
     public abstract boolean isFutureVersion();
 
+    /**
+     * Returns this build version in a form suitable for storing in node metadata
+     */
+    public abstract String toNodeMetadata();
```

Comment:
Why is the string version specific to node metadata? Could it just be toString and fromString?

Reply:
The default `toString` is the version semver, not the id number